app/component/search/search-two-fields-container.cjsx

```diff
@@ -15,6 +16,24 @@ class SearchTwoFieldsContainer extends React.Component
   constructor: ->
     super
 
+    origin = @context.getStore('EndpointStore').getOrigin()
+    dest = @context.getStore('EndpointStore').getDestination()
+
+    @state =
```

Comment:
why not

```
@state =
  origin:
    userInput: !origin.useCurrentPosition
  dest:
    userInput: !dest.useCurrentPosition
  
```

Also, please use destination rater than dest, as it is used elsewhere in the file